qiskit/circuit/__init__.py

```diff
@@ -10,38 +10,64 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""
-========================================
-Quantum Circuits (:mod:`qiskit.circuit`)
-========================================
+r"""
+=============================================
+Quantum circuit model (:mod:`qiskit.circuit`)
+=============================================
 
 .. currentmodule:: qiskit.circuit
 
-Overview
-========
+The fundamental element of quantum computing is the *quantum circuit*.  This is a computational
+routine that can be run, one shot at a time, on a quantum processing unit (QPU).  A circuit will act
+on a predefined amount of quantum data (in Qiskit, we only directly support qubits) with unitary
+operations (gates), measurements and resets.  In addition, a quantum circuit can contain operations
+on classical data, including real-time computations and control-flow constructs, which are executed
+by the controllers of the QPU.
+
+.. note::
+
+    You may wish to skip the introductory material and jump directly to:
+
+    * :ref:`the API overview of the whole circuit module <circuit-module-api>`
+    * :ref:`the detailed discussion about how circuits are represented <circuit-repr>`
+    * the core :class:`QuantumCircuit` class for how to build and query circuits
+    * :ref:`information on construction custom instructions <circuit-custom-gates>`
+    * :ref:`ways to work with circuit-level objects <circuit-working-with>`
+    * :ref:`discussion of Qiskit conventions for circuits, matrices and state labelling
+      <circuit-conventions>`
+
+Circuits are at a low level of abstraction when building up quantum programs.  They are the
+construct that is used to build up to higher levels of abstraction, such as the :ref:`primitives of
+quantum computation <qiskit-primitives>`, which accumulate bitwise data from many shots of
```

Comment:
> accumulate bitwise data

I think the "bitwise" qualifier here is unnecessary. A circuit in the future may return data following not `bit[n]` types, including `int`s, `angle`s, `complex`, etc. The sentence still works if you drop "bitwise".

Reply:
I'm ok to remove "bitwise" here if you still prefer, just a small argument in its defence first:

A circuit may return data that's not a `bit` type, but that data always would have been built up by switching on data that _was_ - that's the point I'm trying to emphasise.  I didn't see (say) IPE as being contrary to "bitwise"; the `angle` is still built up in a single circuit execution as bit-by-bit operations, and error statistics or smoothing of all those single-shot measurements would be the domain of a primitive.  For a `complex`, say, a single circuit execution can only build that up off single-bit measurements (however many of them), or things like `if` statements based on those that set the value.

Ofc, I suppose you could make the same argument about the primitives as well, it's just the "level" that the bitwise stuff happens is even further remove.

Reply:
I was thinking of `complex` arising as the outcome of using the future Qiskit equivalent of OpenPulse's `measure_iq` which produces a single complex value as opposed to a single bit.

Reply:
And I definitely was thinking about IPE when suggesting `angle` as a possible return type.

Reply:
Yeah, the OpenPulse enhanced measurements are a good point, and you could say the result of an `extern` call that's stateful in whatever way as well could easily give you structured data within the circuit.

I'll delete the word "bitwise" - I can totally buy that argument.

Reply:
Done in 2f4d5417741.